### ISSUE TITLE
Upgrade @theia/monaco-editor-core (currently 0.20) to the latest Monaco (currently standalone/0.23.x)

### ISSUE THREAD

**Issue by danarad05** [eclipse-theia/theia#8969]
Currently these feature are missing from Theia monaco-editor-core (obviously could be other features as well):
1. ["in" operator for when clauses](https://code.visualstudio.com/updates/v1_49#_in-operator-for-when-clauses) was added in v1.49 (ticket #8691 needs it)
2. vscode URI utils functions like joinPath (ticket #8752 needs it)

- VS Code standalone/0.23.x contains joinPath function:
   https://github.com/microsoft/vscode/blob/standalone/0.23.x/src/vs/base/common/uri.ts
- VS Code standalone/0.20.x does not contain joinPath function:
   https://github.com/microsoft/vscode/blob/standalone/0.20.x/src/vs/base/common/uri.ts

References:
Migration basic Guidelines - https://github.com/eclipse-theia/theia/wiki/LSP-and-Monaco-integration#migration-guidelines
#5412 (comment) - althought I think that is outdated as current repo is @theia/monaco-editor-core and not TypeFox/monaco-editor-core.

**Comment by amiramw** [comment]
Previous upgrade for reference: https://github.com/eclipse-theia/theia/pull/8010

**Comment by azatsarynnyy** [comment]
Thank you @danarad05 for registering the issue.
I believe @RomanNikitenko can share his knowledge and experience he has got during a couple of the latest Monaco upgrades.
It would be useful for anyone who will decide to do the subsequent upgrades.
It would be great to have all the required steps described somewhere on the wiki, maybe update the existing [Migration Guidelines](https://github.com/eclipse-theia/theia/wiki/LSP-and-Monaco-integration#migration-guidelines).

**Comment by RomanNikitenko** [comment]
The main required steps for Monaco upgrade:
- publish a new version of `@theia/monaco-editor-core` and apply this dependency for `theia`, please see details here: https://github.com/eclipse-theia/theia/issues/7697#issuecomment-638011680
- the upgrade process has been greatly simplified by removing `monaco-languageclient` dependency, so we don't need upgrade `monaco-languageclient` [like here](https://github.com/TypeFox/monaco-languageclient/pull/207) this time, but maybe some dependencies related to Monaco should be updated like: `vscode-ws-jsonrpc`
- [signatures](https://github.com/eclipse-theia/theia/blob/master/packages/monaco/src/typings/monaco/index.d.ts) alignment (signatures exist and matching in a new version of Monaco) - I provided references to the corresponding types in VS Code at the previous Monaco upgrade - it should simplify the process of searching the corresponding types
- [editor configurations](https://github.com/eclipse-theia/theia/blob/master/packages/editor/src/browser/editor-preferences.ts) alignment, please see [instructions](https://github.com/eclipse-theia/theia/blob/4a3e133bfdab7076dd0290a59f5523c85673d3f2/packages/editor/src/browser/editor-preferences.ts#L66-L74)
- testing/fixing. Some time ago @akosyakov provided integration tests: https://github.com/eclipse-theia/theia/pull/7265. It helps to test the changes 


@azatsarynnyy 
> It would be great to have all the required steps described somewhere on the wiki, maybe update the existing Migration Guidelines.

I described the required steps above, I'm going to add some items within the next upgrade if I missed something. Also it will be the first Monaco upgrade after removing `monaco-languageclient` dependency. 
I think within coming Monaco upgrade we could go through the provided steps ^ first and then update the corresponding wiki page.
WDYT?

**Comment by azatsarynnyy** [comment]
> I think within coming Monaco upgrade we could go through the provided steps ^ first and then update the corresponding wiki page.
> WDYT?

Agree :+1:

**Comment by amiramw** [comment]
Assigned this to @danarad05. If someone else started looking into it, please let us know.

**Comment by danarad05** [comment]
@RomanNikitenko 
Hi Roman, 
1. I'm not sure I understand the first step here - currently there isn't a `standalone/0.22.x` branch in `theia-ide/vscode`. 
Could you elaborate what are prior steps to this first step?
2. Also, what version is relevant for standalone/0.22.x? https://github.com/microsoft/vscode/releases/tag/1.53.2?
 
![image](https://user-images.githubusercontent.com/71069170/107932404-332fbf00-6f86-11eb-91b7-04481fdb818c.png)

**Comment by RomanNikitenko** [comment]
@danarad05 
Hello!

https://github.com/theia-ide/vscode was forked from https://github.com/microsoft/vscode
so you should fetch the corresponding branch from the fork first.

**Comment by danarad05** [comment]
@RomanNikitenko 
fetched from upstream = microsoft/vscode, updated the disable treeshake and monaco package.json but couldn't push changes - don't have required permissions. 
Please advise,
Thanks

![image](https://user-images.githubusercontent.com/71069170/107939929-21ebb000-6f90-11eb-9e18-1e5bcd79d804.png)

**Comment by RomanNikitenko** [comment]
@danarad05 
I checked and faced with the same permission error

![permissions_error](https://user-images.githubusercontent.com/5676062/107948050-fc64a380-6f9b-11eb-84d3-9d5f8f5b6cdc.png)

@akosyakov please take a look https://github.com/eclipse-theia/theia/issues/8969#issuecomment-779154231

**Comment by RomanNikitenko** [comment]
@kittaakos @marcdumais-work @vince-fugnitto 

Maybe someone knows who has the required permissions to publish a new version of `@theia/monaco-editor-core`. 

The problem is: at the current step @danarad05 can not push the changes to https://github.com/theia-ide/vscode.
The next step - publishing a new version of `@theia/monaco-editor-core`.   
Please see the comments above.

Thanks in advance!

**Comment by marcdumais-work** [comment]
@RomanNikitenko I have just given @danarad05 write access to that repo - just need to accept GH's invite,

**Comment by marcdumais-work** [comment]
(and you too @RomanNikitenko, in case you need it)

**Comment by akosyakov** [comment]
@RomanNikitenko You will need to fine someone else to review. I don't have time for Theia recently.

**Comment by danarad05** [comment]
> but maybe some dependencies related to Monaco should be updated like: `vscode-ws-jsonrpc`
@RomanNikitenko 
I don't see any usage of vscode-ws-jsonrpc on standalone/0.22.x. Do you mean to add it to 0.22.x use it instead of exisiting WebSocket?

**Comment by RomanNikitenko** [comment]
@danarad05 
I think we don't need any other changes for `standalone/0.22.x` of https://github.com/theia-ide/vscode except https://github.com/theia-ide/vscode/commit/863cb53b25c20588322abbb896b6a6e1a8e93fa2.

About 
> the upgrade process has been greatly simplified by removing monaco-languageclient dependency, so we don't need upgrade monaco-languageclient like here this time, but maybe some dependencies related to Monaco should be updated like: vscode-ws-jsonrpc

I meant that probably we have to update some dependencies **on https://github.com/eclipse-theia/theia side**. 
The next step is publishing a new version of `@theia/monaco-editor-core` to the `npm` registry and applying it to `theia`.

**Comment by danarad05** [comment]
@RomanNikitenko 

Not able to publish to npmjs. Please advise.

> npm notice === Tarball Details ===
npm notice name:          @theia/monaco-editor-core
npm notice version:       0.22.3
npm notice package size:  8.1 MB
npm notice unpacked size: 43.1 MB
npm notice shasum:        2de06f18b8dd3461786b53ab9a7a6282808063bb
npm notice integrity:     sha512-VmylSNp2/F0F3[...]8nssT0snEPZQw==
npm notice total files:   663
npm notice
npm ERR! code E404
npm ERR! 404 Not Found - PUT https://registry.npmjs.org/@theia%2fmonaco-editor-core - Not found
npm ERR! 404
npm ERR! 404  '@theia/monaco-editor-core@0.22.3' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\C5313341\AppData\Roaming\npm-cache\_logs\2021-02-18T09_21_31_859Z-debug.log

**Comment by RomanNikitenko** [comment]
@danarad05 
I don't know the details, but can be related to https://stackoverflow.com/questions/39115101/getting-404-when-attempting-to-publish-new-package-to-npm

**Comment by danarad05** [comment]
@RomanNikitenko 
@amiramw 
I logged into `npm` successfully before `npm publish` and got this error. I tried now various things but still didn't manage to publish successfully.

**Comment by marcdumais-work** [comment]
Hi @danarad05 ,

> Are the permissions needed to be able to publish to that npm package?

I think so. Did someone provide you with an npm publish token for @theia? If not you probably need one.

**Comment by RomanNikitenko** [comment]
I don't have `an npm publish token for @Theia` - at the previous Monaco upgrade Anton published `@theia/monaco-editor-core` to the npm registry. 

AFAIK @danarad05  prepared the corresponding branch for the publishing.
So, someone, who has the token, could help @danarad05 with it - publish it using the branch or provide the token.

**Comment by marcdumais-work** [comment]
One option would be to add the publish token as a secure variable or whatever it's called and publish from a GitHub action. WDYT?

**Comment by marcdumais-work** [comment]
BTW, would it make sense for that repo to be hosted at the Foundation? Similarly to https://github.com/eclipse-theia/vscode-builtin-extensions?

**Comment by RomanNikitenko** [comment]
> One option would be to add the publish token as a secure variable or whatever it's called and publish from a GitHub action. WDYT?

Sounds good for me!

@benoitf @azatsarynnyy 
please share your opinion about:
- https://github.com/eclipse-theia/theia/issues/8969#issuecomment-781807636
- https://github.com/eclipse-theia/theia/issues/8969#issuecomment-781813188

**Comment by azatsarynnyy** [comment]
> One option would be to add the publish token as a secure variable or whatever it's called and publish from a GitHub action. WDYT?

you mean a GitHub secret? sounds good :+1:

**Comment by benoitf** [comment]
yes I like the idea to be hosted on the foundation repositories with a github action/github secret automatically publishing stuff (or on demand workflow)

**Comment by danarad05** [comment]
> BTW, would it make sense for that repo to be hosted at the Foundation? Similarly to https://github.com/eclipse-theia/vscode-builtin-extensions?

@marcdumais-work 
I see that @benoitf @azatsarynnyy @RomanNikitenko agree with you. Who/how is that done?
Thanks

**Comment by marcdumais-work** [comment]
@danarad05 I have set a npm publish token as a secret in repo `theia-ide/vscode`: `NPM_TOKEN`. We'll need to test it. 

Here's an example of a GH Workflow where a token is used to publish to npm:
https://github.com/eclipse-theia/theia/blob/master/.github/workflows/build.yml

And another one where we use a [magic branch](https://github.com/eclipse-theia/vscode-builtin-extensions/blob/master/.github/workflows/publish-vsx-specific-latest.yml#L5), and publish whatever is pushed on that branch:
 https://github.com/eclipse-theia/vscode-builtin-extensions/blob/master/.github/workflows/publish-vsx-specific-latest.yml#L12

I think we could setup something similar for Monaco: we just need to pick a magic branch name and give write access to the correct people, and add a workflow that activates on that branch, builds and publishes @theia/monaco.

**Comment by marcdumais-work** [comment]
About bringing the repo to the foundation, I think we'll need to see with them what they think, given the nature of the repo and what's done with it. i.e. I am not sure the usual "re-license and move to the Foundation process works in this case, since it's a 3rd party repo rather than one of ours.

**Comment by marcdumais-work** [comment]
The email discussion may not be over, but the suggestion so far is to open a "type B" CQ to validate the Monaco content we "fork", so that the it will be fully reviewed (not just lighter "type A" process that checks for license compatibility).

Here's an example:  the CQ I opened for `vscode-builtin-extensions`, that can serve as a guide: 
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22943

We may not have to relicense or sublicense the repo either, from what I now understand.

**Comment by marcdumais-work** [comment]
A project committer needs to open the CQ, since only them can access I think.

**Comment by azatsarynnyy** [comment]
> A project committer needs to open the CQ, since only them can access I think.

I can see @amiramw among the committers [here](https://projects.eclipse.org/projects/ecd.theia/who), from the SAP side, but not @danarad05 (yet :wink:). So, @amiramw could you look at the CQ?
@marcdumais-work I think we need to include @amiramw to the mail thread as well.

**Comment by danarad05** [comment]
@marcdumais-work FYI, it seems that there is already version standalone/0.23,x@vscode.

**Comment by marcdumais-work** [comment]
> @marcdumais-work FYI, it seems that there is already version standalone/0.23,x@vscode.

Sorry - I miss context to understand. Do you mean that we should target that new version instead of standalone/0.22.x? 

I think we can proceed without waiting for the repo to me moved to the Foundation, which can happen independently. See [above](https://github.com/eclipse-theia/theia/issues/8969#issuecomment-784332832) for how I suggest to proceed.

**Comment by danarad05** [comment]
Created new PR for standalone/0.23.x: #9154

**Comment by danarad05** [comment]
>Sorry - I miss context to understand. Do you mean that we should target that new version instead of standalone/0.22.x?

I created 0.23.x versions.

> And another one where we use a [magic branch](https://github.com/eclipse-theia/vscode-builtin-extensions/blob/master/.github/workflows/publish-vsx-specific-latest.yml#L5), and publish whatever is pushed on that branch:
> https://github.com/eclipse-theia/vscode-builtin-extensions/blob/master/.github/workflows/publish-vsx-specific-latest.yml#L12
> 
> I think we could setup something similar for Monaco: we just need to pick a magic branch name and give write access to the correct people, and add a workflow that activates on that branch, builds and publishes @theia/monaco.

Added a [workflow](https://github.com/theia-ide/vscode/blob/standalone/0.23.x/.github/workflows/publish-monaco-editor-core-to-npmjs.yml) under standalone/0.23.x. 
Who can check if correct and how do we set up a "magic" branch (I named it `monaco-editor-core-publish` in yml)? 

Thanks

**Comment by marcdumais-work** [comment]
Hi @danarad05 

> Added a [workflow](https://github.com/theia-ide/vscode/blob/standalone/0.23.x/.github/workflows/publish-monaco-editor-core-to-npmjs.yml) under standalone/0.23.x.

Seems ok at a quick look

> Who can check if correct and how do we set up a "magic" branch (I named it `monaco-editor-core-publish` in yml)?

I do not think more setup is required at this time (we could optionally consider to "protect" the branch). To test this, I suggest you temporarily setup your branch to publish a `next` [version of the package](https://github.com/theia-ide/vscode/commit/c2207421db74a0d3ef51e5ef073359691b9bb7c8#diff-006d75594e1ee9304dfb0dd31433a6c7b04a64c58dd72a4fd6afb449099cdcd2R4) and then push your branch to `monaco-editor-core-publish`. If everything works, the `next` should successfully publish to `npm` and can be used to test the feature. 

Once the PR is approved, you would similarly push the PR branch to the magic branch, to publish the solid version of the package.

**Comment by danarad05** [comment]
Hi @marcdumais-work 

> To test this, I suggest you temporarily setup your branch to publish a `next` [version of the package](https://github.com/theia-ide/vscode/commit/c2207421db74a0d3ef51e5ef073359691b9bb7c8#diff-006d75594e1ee9304dfb0dd31433a6c7b04a64c58dd72a4fd6afb449099cdcd2R4) and then push your branch to `monaco-editor-core-publish`. If everything works, the `next` should successfully publish to `npm` and can be used to test the feature.

Tried to publish to npm but to no avail. Are you sure token you set is correct? if not that I'm not sure what causes failure. Could you take a look at last [run](https://github.com/theia-ide/vscode/runs/2106984425?check_suite_focus=true)?

**Comment by danarad05** [comment]
@marcdumais-work I'm in need of your help - or someone with relevant privileges and knowledge. 
https://github.com/eclipse-theia/theia/issues/8969#issuecomment-799218647

**Comment by marcdumais-work** [comment]
Hi @danarad05 ,

Sorry for the delay. Let me have a look now. If that doesn't work soon we can find a way to communicate securely and I can provide you with a personal publish token.

**Comment by marcdumais-work** [comment]
Dan, I think maybe you GH workflow is missing something like this:
https://github.com/eclipse-theia/theia/blob/master/.github/workflows/build.yml#L117

Adapted to the token's name in this repo, I think we need:
```yml
env:
   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

**Comment by danarad05** [comment]
> Dan, I think maybe you GH workflow is missing something like this:
> https://github.com/eclipse-theia/theia/blob/master/.github/workflows/build.yml#L117
> 
> Adapted to the token's name in this repo, I think we need:
> 
> ```yaml
> env:
>    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
> ```

Please check latest: https://github.com/theia-ide/vscode/actions/runs/661035621/workflow

![image](https://user-images.githubusercontent.com/71069170/111498650-cbb68c00-874a-11eb-955a-e90142b4f0f5.png)

**Comment by marcdumais-work** [comment]
Then the token itself seems to be the most likely cause. I can generate and set a fresh one. 

@theia/monaco-editor-core@0.23.0

> BTW, the branch you attempted to publish would have published a solid revision of the component

The version here needs to be modified: 
https://github.com/theia-ide/vscode/blob/monaco-editor-core-publish/build/monaco/package.json#L3

I suggest: '0.23.0-next-${SHA}'
Where SHA is the vscode commit you branched-off from. Seems to be: 
SHA=82e8ea39fc101d63

**Comment by danarad05** [comment]
> BTW, the branch you attempted to publish would have published a solid revision of the component

I figured I could unpublish in worse case it worked :).
Thanks.

**Comment by marcdumais-work** [comment]
Dan, I have updated the token. Do you want to try again?

**Comment by marcdumais-work** [comment]
I have re-started the latest workflow, with your changes to publish a next version.

**Comment by marcdumais-work** [comment]
Success: https://www.npmjs.com/package/@theia/monaco-editor-core/v/0.23.0-next-82e8ea39fc101d63

**Comment by marcdumais-work** [comment]
> BTW, the branch you attempted to publish would have published a solid revision of the component, which we are not to do until we open and have approved the CQ we discussed about.
> 
> @amiramw will you take care of it?

Correction: as previously discussed we did not want to couple this Monaco upgrade with the process of bringing this repo/component under our EF project. 

So @danarad05 when you and the committers helping you think it's ready for release, just push that version, with solid revision, to the magic branch. Maybe you can use the published "next" version in the Theia framework side of this upgrade, in case further changes are required before you're done.

@amiramw We should start the "move" process ASAP, even if finishing is not a pre-requisite for this upgrade. We have Wayne's email as a guide.

**Comment by danarad05** [comment]
> Success: https://www.npmjs.com/package/@theia/monaco-editor-core/v/0.23.0-next-82e8ea39fc101d63

Hallelujah @marcdumais-work .. Merci beaucoup! at last!

> Maybe you can use the published "next" version in the Theia framework side of this upgrade

Yes, I have to use it in #9154 

Thanks

**Comment by marcdumais-work** [comment]
Dan, 

Some things I think we need to do starting ASAP: 

We would massage that `package.json`. At least change the `homepage` and `repository` to point to our repo rather than Microsoft's. Else a reasonable person might think that Microsoft makes and supports this `npm` package, open bugs against their repos, etc. It's only fair.

I think we also need to customise the README, to make it clear what this package is, rather than keep the [default one](https://www.npmjs.com/package/@theia/monaco-editor-core/v/0.23.0-next-82e8ea39fc101d63). We need to make sure we bundle the license (probably copy it from vscode's root folder) and refer to it in `package.json`. 

As a guide, please see what we've [done](https://github.com/eclipse-theia/vscode-builtin-extensions/blob/master/src/package-vsix.js#L131-L136) for `vscode-builtin extensions` . E.g. see this one's readme and repository link that points to our repo: https://open-vsx.org/extension/vscode/typescript-language-features.

**Comment by RomanNikitenko** [comment]
Hello, Dan!
I haven't bumped into the problem at the previous upgrades.
So, unfortunately, I have no idea what's the problem...

**Comment by marcdumais-work** [comment]
@eclipse-theia/core can anyone help Dan with the issue [above](https://github.com/eclipse-theia/theia/issues/8969#issuecomment-806929278)?

**Comment by kittaakos** [comment]
> compiled to `es5` like Theia?

As far as I know yes. See  https://github.com/eclipse-theia/theia/issues/5902 and https://github.com/eclipse-theia/theia/issues/6761.

**Comment by danarad05** [comment]
@RomanNikitenko  @kittaakos 

> Maybe you can try to tweak it somewhere here:

I tried a few things there but was unsuccessful. 

After looking further in this, I think we might have an issue here as `standalone/0.23.x` is configured to compile to `es6` while `0.20.x` is set to `es5`:

![image](https://user-images.githubusercontent.com/71069170/112830848-c7b83180-909b-11eb-8b9f-350f5bdb06ce.png)

If I try to downgrade compilation to es5 - I'm getting these errors:
![image](https://user-images.githubusercontent.com/71069170/112837884-1a4a1b80-90a5-11eb-9549-097ae4c70153.png)

Possible solutions as I see currently:
1. Change `MonacoEditorService` to not extend `monaco.services.CodeEditorServiceImpl` - but use it internally? (**currently!** only `MonacoEditorService` seems problematic)
2. Overwrite 3 problematic instances above for ES5 compliance - and compile 0.23.x to `es5` still?
3. Compile Theia with es6 - probably major change?

Please advise!

Thanks

**Comment by danarad05** [comment]
> 1. Change `MonacoEditorService` to not extend `monaco.services.CodeEditorServiceImpl` - but use it internally? (**currently!** only `MonacoEditorService` seems problematic)

- I changed `MonacoEditorService` to not extend `monaco.services.CodeEditorServiceImpl` and it worked but now getting error _Class constructor ResolvedKeybinding cannot be invoked without 'new'_ - so this means that all classes extending `monaco.*` will need to be updated - so not a great solution.

**Comment by marcdumais-work** [comment]
@eclipse-theia/ecd-theia-committers can anyone can help Dan?

**Comment by RomanNikitenko** [comment]
Dan, unfortunately I have no experience in the mentioned above problem - so - sorry - I can not provide any help/advise.

**Comment by azatsarynnyy** [comment]
If I understood correctly from today's Theia Dev meeting, @JonasHelming's team is going to look at the ES6 migration issue.

**Comment by danarad05** [comment]
Thank you @marcdumais-work @RomanNikitenko @azatsarynnyy. Unfortunately I couldn't attend this week's meeting. 

Also, I might have managed to compile `standalone/0.23.x` to `ES5` successfully - with doing some changes to code (trying to do minimal changes) - still checking -  and still it doesn't mean to that is way to go - but wanted to see if possible. 

For instance - compile completed successfully although TS file shows TS problem:
![image](https://user-images.githubusercontent.com/71069170/113020532-5c4d8d00-918b-11eb-9506-f0913d521949.png)

Not sure yet if `Object.values()` is actually working - will try to test it tomorrow:
![image](https://user-images.githubusercontent.com/71069170/113021065-dda51f80-918b-11eb-98e2-559a0973b3c3.png)

**Comment by danarad05** [comment]
@marcdumais-work @RomanNikitenko @azatsarynnyy @offer8

Please check changes I've made to `@theia/monaco-editor-core` for compiling to ES5: https://github.com/theia-ide/vscode/commit/07fb6e15d8e6b31c93de11b03da03cc6e569d64f

Seems to me as changes we could live with (?) - if all is working properly. If there are objections for this route - please let me know now. 
I have also published it to `npmjs.org` as `0.23.0-next-82e8ea39fc101d63.2` version - in order to be able to consume it in Theia.

I will now test it in Theia and let you know

**Comment by marechal-p** [comment]
@danarad05 rather than modifying the TS sources should you transpile the JS output from ES6 to ES5?

This is what @akosyakov had done if I understand correctly: https://github.com/eclipse-theia/theia/blob/71f762c59580670a3ae1f661bc10010f0d5581f1/dev-packages/application-manager/src/generator/webpack-generator.ts#L177-L196

**Comment by danarad05** [comment]
@marechal-p 
The code you quoted is from Theia. I'm talking about `@theia/monaco-editor-core` there there isn't usage of babel

**Comment by marechal-p** [comment]
I quoted a Theia configuration that is used to consume ES6. If Theia can consume ES6 that way maybe you don't need to change monaco's target from ES6 to ES5 then.

**Comment by danarad05** [comment]
> Since we already transpile in Theia

1. We don't transpile `node_modules` and so the monaco.* interfaces that we defined in `Theia` are referencing `@theia/monaco-editor-core` library that is transpiled to `ES6` - which causes initialization issues like this:
![image](https://user-images.githubusercontent.com/71069170/112496569-84a14a00-8d8d-11eb-8d84-f16f088ffe4e.png).

2. I tried before doing these changes (https://github.com/theia-ide/vscode/commit/07fb6e15d8e6b31c93de11b03da03cc6e569d64f) to add babel and was unsuccessful with it - it still gave me errors and wouldn't complete compilation successfully. Don't know if I did it wrong and if there is a way to do it that successfully..? 

In any case - even if we can refrain for the changes I've done, we still need to transpile to ES5 the library as it won't be transpiled by Theia (as long as Theia is configured for ES5).

**Comment by caseyflynn-google** [comment]
Hey All! 

Is there any update on this? We are hoping this will resolve https://github.com/eclipse-theia/theia/issues/8752 so we may update the Go VS Code extension in our deployment. Is there anything we can do to assist?

**Comment by danarad05** [comment]
Hi @caseyflynn-google 

I hope that in a few days/week we'll have a complete solution which consists of 2 parts:
1) First part (this issue) `@theia/monaco-editor-core` upgraded to 0.23.x - which is already ready.
2) Second part #9154 - which are changes required in Theia itself - which includes replacing `quickOpen` modules with `quickInput`  modules - as 0.23.x does not support `quickOpen` anymore.

When #9154 will be completed we would need a lot of testing as this is a large refactoring. So your help with that would be appreciated.

Thanks

**Comment by azatsarynnyy** [comment]
@danarad05 from the Red Hat side, we have a couple of people to help with testing and reviewing the contribution.
Just to let you know in case help is needed.

**Comment by danarad05** [comment]
@caseyflynn-google @westbury 
cc: @marcdumais-work 

> Is there any update on this? We are hoping this will resolve #8752 so we may update the Go VS Code extension in our deployment. Is there anything we can do to assist?

PR #9154 is ready for review / testing - if you would like to join in the effort - you are welcome.

Thanks

**Comment by tsmaeder** [comment]
Reopening....I have a follow-up refactoring we decided to do post-merge of the first PR.

**Comment by tsmaeder** [comment]
Followup refactoring now merged.
